tools/autograd/templates/Functions.cpp

```diff
@@ -425,6 +425,16 @@ Tensor cumsum_backward(const Tensor & x, int64_t dim) {
   return ret;
 }
 
+Tensor cummax_backward(const Tensor &grad, const Tensor &input, int64_t dim) {
+  if (input.dim() == 0) {
+    return input;
+  }
+  auto result = at::native::cummax(input, dim);
```

Comment:
No need to recompute this, you have access to the result of the forward already by changing the `derivatives.yaml` to `self: cummax_backward(grad.to(self.scalar_type()), self, cummax_out, cummax_indices, dim)`.

Reply:
good catch! will do